ingestion/src/metadata/ingestion/source/dashboard/powerbi/metadata.py

```diff
@@ -360,18 +362,36 @@ def yield_bulk_datamodel(
         Method to fetch DataModels in bulk
         """
         try:
-            data_model_request = CreateDashboardDataModelRequest(
-                name=EntityName(dataset.id),
-                displayName=dataset.name,
-                description=Markdown(dataset.description)
-                if dataset.description
-                else None,
-                service=FullyQualifiedEntityName(self.context.get().dashboard_service),
-                dataModelType=DataModelType.PowerBIDataModel.value,
-                serviceType=DashboardServiceType.PowerBI.value,
-                columns=self._get_column_info(dataset),
-                project=self._fetch_dataset_workspace(dataset_id=dataset.id),
-            )
+            if isinstance(dataset, Dataset):
```

Comment:
nit - if we expect data to be dataset or dataflow here, let's also change the function signature with a Union[bla, bla].

also, it's ok to instance checks on Dataset and Dataflow both, and if we get something that's neither, raise an exception